blocks/ifus/ifus.js

```diff
@@ -0,0 +1,248 @@
+import { translate } from '../../scripts/lib-franklin.js';
+
+function createDomStructure(structure, parentElement = document.body) {
+  structure.forEach((element) => {
+    const domElement = document.createElement(element.type);
+    if (element.attributes) {
+      Object.keys(element.attributes).forEach((attr) => {
+        domElement.setAttribute(attr, element.attributes[attr]);
+      });
+    }
+
+    if (element.textContent) {
+      domElement.textContent = element.textContent;
+    }
+
+    if (element.children) {
+      createDomStructure(element.children, domElement);
+    }
+
+    if (element.classes) {
+      element.classes.forEach((c) => domElement.classList.add(c));
+    }
+
+    if (element.position === 'prepend') {
+      parentElement.prepend(domElement);
+    } else {
+      parentElement.appendChild(domElement);
+    }
+  });
+}
+
+function populateSearch(
```

Comment:
should be `async` due to the use of `translate`

Reply:
I don't think it needs to be `async` as the `translate` is only used in the event listener callback (which is async). However, I agree that is somewhat hard to parse - hence, I moved the callback into a separate function so that it is easier to see where the `async` part happens.